hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieActiveTimeline.java

```diff
@@ -676,6 +688,9 @@ private void createImmutableFileInPath(Path fullPath, Option<byte[]> content) {
         if (null != fsout) {
           fsout.close();
         }
+        if (null != tmpPath) {
+          metaClient.getFs().rename(tmpPath, fullPath);
```

Comment:
Are renames atomic on S3 or other cloud object storage? I know put and delete in s3 is strongly consistent but what about copy?

Reply:
Yes we cannot rename :) is there a way to do this specific to hdfs?

Reply:
Can our `ConsistencyGuard` guarantee this?